src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -484,12 +491,11 @@ private BulkResponse execute() throws Exception {
             if (reporter != null) {
               for (R record : batch) {
                 SinkRecord original = recordsToReportOnError.get(record);
-                if (original != null) {
+                BulkResultItem result = bulkRsp.failedRecords.get(record);
```

Comment:
can we add a javadoc in `BulkResponse` constructor that says that this can't be `null`(never `null`) but can be empty?